File: engine/src/main/java/com/trading/hf/core/PriceRegistry.java
package com.trading.hf.core;

import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;

public class PriceRegistry {
    private static final Map<String, Double> latestPrices = new ConcurrentHashMap<>();

    public static void updatePrice(String symbol, double price) {
        latestPrices.put(symbol, price);
    }

    public static double getPrice(String symbol) {
        return latestPrices.getOrDefault(symbol, 0.0);
    }
}

File: python_engine/core/price_registry.py
from typing import Dict

class PriceRegistry:
    _prices: Dict[str, float] = {}

    @classmethod
    def update_price(cls, symbol: str, price: float):
        cls._prices[symbol] = price

    @classmethod
    def get_price(cls, symbol: str) -> float:
        return cls._prices.get(symbol, 0.0)
